packages/node_modules/@node-red/nodes/core/common/20-inject.html

```diff
@@ -201,27 +223,27 @@
             }
             if (this.name) {
                 return this.name+suffix;
-            } else if (this.payloadType === "string" ||
-                    this.payloadType === "str" ||
-                    this.payloadType === "num" ||
-                    this.payloadType === "bool" ||
-                    this.payloadType === "json") {
-                if ((this.topic !== "") && ((this.topic.length + this.payload.length) <= 32)) {
-                    return this.topic + ":" + this.payload+suffix;
-                } else if (this.payload.length > 0 && this.payload.length < 24) {
-                    return this.payload+suffix;
+            } else if (payloadType === "string" ||
+                    payloadType === "str" ||
+                    payloadType === "num" ||
+                    payloadType === "bool" ||
+                    payloadType === "json") {
+                if ((topic !== "") && ((topic.length + payload.length) <= 32)) {
+                    return topic + ":" + payload+suffix;
+                } else if (payload.length > 0 && payload.length < 24) {
+                    return payload+suffix;
                 } else {
                     return this._("inject.inject")+suffix;
                 }
-            } else if (this.payloadType === 'date') {
-                if ((this.topic !== "") && (this.topic.length <= 16)) {
-                    return this.topic + ":" + this._("inject.timestamp")+suffix;
+            } else if (payloadType === 'date' || payloadType === 'bin' || payloadType === 'env') {
```

Comment:
bin & env payload types did not render a label in the old version. This change allows all three (date, bin & env) to render a smart label.